### PR TITLE
Add support for HackerNews, Imgur and Steam

### PR DESCRIPTION
**Pull request by streeck**

### CODE REVIEW

File: providers/hackernews/hackernews.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+func init() {
+	providers.AddProvider("hackernews", &Hackernews{})
+}
+
+// Hackernews adheres to the Provider interface.
+type Hackernews struct {
+}
+
+// BuildURI generates a search URL for Amazon.
```
Comment (KeizerDev) [inline providers/hackernews/hackernews.go]: Comment says Amazon

Comment (streeck) [inline providers/hackernews/hackernews.go]: Oops, my bad, forgot that one. Will fix it.

**Comment by KeizerDev** [comment]
@streeck can you merge 8f24810 into 89d2c93?

**Comment by streeck** [comment]
Sure, I've squashed the commits and pushed the changes. Together with Imgur and Steam additions.

**Comment by zquestz** [comment]
Looks like we have a minor conflict in this PR. Probably just the README. When the button goes green I am happy to merge!

**Comment by KeizerDev** [comment]
@zquestz button is green for me :stuck_out_tongue_winking_eye:

**Comment by streeck** [comment]
Thanks, also, I'm not sure if I dealt with the conflict the right way. I fetch the latest changes from your repository to my fork and then merged resolving my additions conflicting with both README.md and s.go.

**Comment by zquestz** [comment]
Just some formatting needed fixing. They were alphabetical. I fixed it for you. =)

**Comment by streeck** [comment]
Ah, I see. Thanks =).

**Comment by zquestz** [comment]
No, thank you for contributing. The least I can do. =)

**Pull request merged**